### PR TITLE
[WIP] [web] More EOS adaptations

### PR DESCRIPTION
**Pull request by dgdavid**
## TL;DR

Start applying EOS suggestions/recommendations to the UI we have produce until now.

## Context

Although we stated in #15 our intention to go for PatternFly as a component library and [EOS as a design system](https://www.eosdesignsystem.com), we didn't start to apply the latest one until now because somehow we were focused in setting the ground to have an [MVP](https://medium.com/anoda-mobile-development-agency/what-is-minimum-viable-product-and-how-to-make-it-right-8b885001c6f5).

So, the goal of this PR is to follow EOS as much as possible. However, there are some stuff that couldn't be applied by now (like no using button as links or the size of headers). They can be addressed in next iterations, along with other things that might be overlooked.

## Important note

This PR adds a dependency because we don't want to rely in an external source for serving the suggested fonts. Instead, we plan to ship them as part of the package.

## In detail (with not-collapsed screenshots)

This sections recaps, adding before and after screenshots, the rules/principles followed here. It can helps reviewers since changes in code by itself doesn't tell too much.

* Applying typography suggestions, read https://suse.eosdesignsystem.com/typography

  | Before | After |
  |-|-|
  | Installation summary before applying typography suggestions | Installation summary after applying typography suggestions |

* Remove dialogs titles (to be concise, read https://eosdesignsystem.herokuapp.com/writing/ux-writing#best-practices) 
  
  | Before | After |
  |-|-|
  | Language selector modal with title | Language selector modal without title, aria-label instead |

  Only one screenshot, but the change applies to all modal dialog (product selector, target selector, etc.)

* Reduce wording of dialogs (https://eosdesignsystem.herokuapp.com/writing/ux-writing#best-practices)

  | Before (verbose) | After (concise) |
  |-|-|
  | Verbose language selector dialog | Concise language selector dialog |
  | Verbose target selector dialog | Concise target selector dialog |

* Do not use variant "link" for action buttons (https://eosdesignsystem.herokuapp.com/buttons#button-vs.-link)

  | Before (using "link" variant) | After (using "secondary" variant) |
  |-|-|
  | Language dialog using link buttons | Language dialog using secondary buttons |
  | Root password dialog using link buttons | Root password dialog using secondary buttons |

  Note that we're still using _variant link_ for two types of buttons:

    * Those that are really a link, even when we are using the `Button` PatternFly component. I.e., those _shortcuts_ from the Installation Summary that allows user to make _quick changes_ for fine-tuning the installation. Maybe we should go for our own `Link` component and stop using buttons for that.
    * The "Do not foo" _non-destructive actions_ used to quickly unset a value, like the root password or the public SSH key.

* Use the SUSE custom focus style to ensure coherent UX across browsers (https://eosdesignsystem.herokuapp.com/focus)
  
  | Before | After |
  |-|-|
  | Installation summary without custom focus style | Installation summary with SUSE custom focus style |

* Do not use the "(see details)" suffix for subvolumes actions links
  
  To follow the "be concise" best practice, https://eosdesignsystem.herokuapp.com/writing/ux-writing#best-practices 

  | | Before | After |
  |-|-|-|
  | Collapsed | Collapsed proposal actions with "see details" suffix | Collapsed proposal actions without the "see details" suffix |
  | Expanded | Expanded proposal actions with "see details" suffix | Expanded proposal actions without "see details" suffix |

### CODE REVIEW

File: web/src/ProductSelector.test.jsx [hunk 1]
```diff
@@ -58,14 +58,16 @@ it("displays the proposal", async () => {
 });
 
 describe("when the user changes the product", () => {
-  it("changes the selected product", async () => {
+  it.only("changes the selected product", async () => {
```
Comment (imobachgs) [inline web/src/ProductSelector.test.jsx:61]: Is this one a left-over?

File: web/src/app.scss [hunk 2]
```diff
@@ -1,24 +1,131 @@
...
-// in development mode)
+// NOTE: eos-base/index imports a lot of styles not needed by now.
+// What is more, EOS is using @import instead of @use, which is the
+// preferred one. Read "What's Wrong With @import?" at
+// https://sass-lang.com/documentation/at-rules/import
+// This block us from importing them granularly as needed because some
+// SCSS EOS modules contains variables from others, making the tools
+// fail. For example, it's the case of titles.scss, which uses a variable
+// from branding.scss. Although we @use it here, the titles.scss
+// cant find the variable because it is namespaced with "branding.".
```
Comment (imobachgs) [inline web/src/app.scss:9]: typo: can't

File: web/src/app.scss [hunk 3]
```diff
@@ -1,24 +1,131 @@
...
-// or "tree-shake" CSS while building (although it might not help while working
-// in development mode)
+// NOTE: eos-base/index imports a lot of styles not needed by now.
+// What is more, EOS is using @import instead of @use, which is the
+// preferred one. Read "What's Wrong With @import?" at
+// https://sass-lang.com/documentation/at-rules/import
+// This block us from importing them granularly as needed because some
+// SCSS EOS modules contains variables from others, making the tools
+// fail. For example, it's the case of titles.scss, which uses a variable
+// from branding.scss. Although we @use it here, the titles.scss
```
Comment (imobachgs) [inline web/src/app.scss:8]: use "@use"? I know, it sounds redundant...

**Review by imobachgs — COMMENTED** [review]
It looks better. Just a few comments.

**Review by joseivanlopez — COMMENTED** [review]
Great summary, as usual! Thanks!

**Review by joseivanlopez — APPROVED** [review]
LGTM
